alphadia/workflow/peptidecentric.py

```diff
@@ -601,54 +425,64 @@ def calibration(self):
                         optimizer.plot()
 
                     break
-
-                features_df, fragments_df = self.extract_batch(
-                    self.optimization_lock_library_precursor_df,
-                    self.optimization_lock_library_fragment_df,
+                self.reporter.log_string(
+                    f"=== Step {current_step}, extracting elution groups {self.optlock.start_index} to {self.optlock.stop_index} ===",
+                    verbosity="progress",
                 )
+                self.optlock.extract()
 
                 self.reporter.log_string(
-                    f"=== Step {current_step}, extracted {len(features_df)} precursors and {len(fragments_df)} fragments ===",
+                    f"=== Step {current_step}, extracted {len(self.optlock.features_df)} precursors and {len(self.optlock.fragments_df)} fragments ===",
                     verbosity="progress",
                 )
 
-                precursor_df = self.fdr_correction(
-                    features_df,
-                    fragments_df,
-                    self.optimization_manager.classifier_version,
-                )
+                self.optlock.fdr()
 
                 self.reporter.log_string(
                     f"=== FDR correction performed with classifier version {self.optimization_manager.classifier_version} ===",
                     verbosity="info",
                 )
 
-                self.log_precursor_df(precursor_df)
+                self.log_precursor_df(self.optlock.precursor_df)
+
+                if self.optlock.reached_target:
```

Comment:
so we need to reach the target twice? think about more defined names here e.g. `reached_n_precursors_target`, `reached_optimization_target` .. ?